### PR TITLE
Hw 4

### PR DESCRIPTION
**Pull request by viskosa**

### CODE REVIEW

File: js-core/homeworks/homework-4/src/main.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  содержащий все объекты содержащие свойство unknownAge: true
+ */
+
+//let array = Array.from({length: 35},
+//  (value, index) => (index % 2 ? {age: index + 2} : {age: NaN}),
+//);
+
+function lookInObject(obj){
+	for (let key in obj) {
+		if (key == 'age' && isNaN(obj[key])) {
```
Comment (OlegLustenko) [inline js-core/homeworks/homework-4/src/main.js]: I would give obj[key] better name it's some abstraction which should be named

File: js-core/homeworks/homework-4/src/main.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+		}
+
+		return obj;
+	}
+}
+
+function makeResultArray(arr) {
+	let resultArray = [];
+	for (let i = 0; i < arr.length; i++) {
+		for (let key in arr[i]) {
```
Comment (OlegLustenko) [inline js-core/homeworks/homework-4/src/main.js]: same as above but about arr[i]

File: js-core/homeworks/homework-4/src/main.js (hunk 1 above)
Comment (viskosa) [reply js-core/homeworks/homework-4/src/main.js]: Should I create some variable for obj[key] ? I thought it is more custom syntax

Comment (viskosa) [reply js-core/homeworks/homework-4/src/main.js]: or maybe did you mean like this:     if (key == 'age' && isNaN(obj.age)

File: js-core/homeworks/homework-4/src/main.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+function FizzBuzz(num) {
+	let res = '';
+	if (num % 3 == 0) {
+		res += 'Fizz';
+	};
+	if (num % 5 == 0) {
+		res += 'Buzz';
+	};
+	if (res.length == 0){
```
Comment (OlegLustenko) [inline js-core/homeworks/homework-4/src/main.js]: I think ternary operator will ideally fit here. What do you think ?

File: js-core/homeworks/homework-4/src/main.js (hunk 1 above)
Comment (OlegLustenko) [reply js-core/homeworks/homework-4/src/main.js]: > Should I create some variable for obj[key] ? I thought it is more custom syntax

yes. It's simply an a element, or something

File: js-core/homeworks/homework-4/src/main.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+ 2. На основании нового массива, создайте новую функцию, которая вернет новый массив
+  содержащий все объекты содержащие свойство unknownAge: true
+ */
+
+//let array = Array.from({length: 35},
+//  (value, index) => (index % 2 ? {age: index + 2} : {age: NaN}),
+//);
+
+function lookInObject(obj){
+	for (let key in obj) {
```
Comment (OlegLustenko) [inline js-core/homeworks/homework-4/src/main.js]: You don't need a loop over the object to find a key 'age'

you can directly access to it

File: js-core/homeworks/homework-4/src/main.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+/*
+ 1. @@SUPER@@. Вам дан массив array, содержащий внутри объекты.
+ Напишите функцию которая внутри цикла проходится по каждому элементу массива
+ И проверяет какой тип данных содержит свойство age, если тип данных NaN,
+ тогда добавляет данному объекту свойство unknownAge: true
+ 2. На основании нового массива, создайте новую функцию, которая вернет новый массив
+  содержащий все объекты содержащие свойство unknownAge: true
+ */
+
+//let array = Array.from({length: 35},
```
Comment (OlegLustenko) [inline js-core/homeworks/homework-4/src/main.js]: that line should be uncommented and then array variable should be used inside your solution.

Not sure if it pushed correct to my repo

File: js-core/homeworks/homework-4/src/main.js [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+	},
+	12,
+	'some string here',
+	{
+		age: NaN,
+		job: 'student'
+	},
+];
+
+console.log(solution(arr));
```
Comment (OlegLustenko) [inline js-core/homeworks/homework-4/src/main.js]: so solution could accept solution(array)

File: js-core/homeworks/homework-4/src/main.js [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+	}
+
+	return resultArray;
+}
+
+function solution(arr) {
+	let newArr = [];
+
+	for (let i = 0; i < arr.length; i++) {
+		if (typeof arr[i] != 'object') continue;
```
Comment (OlegLustenko) [inline js-core/homeworks/homework-4/src/main.js]: if you will use array variable, you don't have to check if array element in object or not, it's always object

File: js-core/homeworks/homework-4/src/main.js (hunk 3 above)
Comment (viskosa) [reply js-core/homeworks/homework-4/src/main.js]: I thought about it)) But I didn't realise what should I write in 'else' case)

File: js-core/homeworks/homework-4/src/main.js (hunk 4 above)
Comment (viskosa) [reply js-core/homeworks/homework-4/src/main.js]: Yes, I can. But I try to make more custom function. But yes, not in this case. It anyway not custom even for this task))

**Comment by viskosa** [comment]
I think I've fixed )

File: js-core/homeworks/homework-4/src/main.js [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+		res += 'Fizz';
+	};
+	if (num % 5 == 0) {
+		res += 'Buzz';
+	};
+	if (res.length == 0){
+		return num;
+	}
+	return res;*/
+	res += (num % 3 == 0) ? 'Fizz' : '';
```
Comment (OlegLustenko) [inline js-core/homeworks/homework-4/src/main.js]: wow :)
first 2 if statements were fine :)

I've thought you would update only last return

Comment (viskosa) [reply js-core/homeworks/homework-4/src/main.js]: But is it ok?

Comment (OlegLustenko) [reply js-core/homeworks/homework-4/src/main.js]: not really, it looks weird. it looks like 
"oh wow never seen such syntax before" :)

**Comment by viskosa** [comment]
fixed again

**Review by OlegLustenko — APPROVED** [review]

**Pull request merged**